mvc/views/razor.rst

```diff
@@ -1,8 +1,283 @@
 
-.. include:: /../common/stub-notice.txt
+Razor is the only view engine that MVC ships, but it is possible to use a
+different view engine.
 
-.. _issue: https://github.com/aspnet/Docs/issues/130
+Implicit Expressions
+^^^^^^^^^^^^^^^^^^^^^
+Implicit Expressions provide a simple mechanism to render data. They are simply
+an ``@`` followed by a variable or method. Razor renders the value to at
+runtime. The following markup renders the variable `` GuestName`` inside an H2
+heading element.
+
+.. code-block:: c#
+
+  <h2>@GuestName</h2>
+
+Razor automatically escapes the ``@`` used in an email address. To escape ``@``
+use ``@@``
+
+*********
+Explicit
+*********
+Sometimes you need to render variables without any space in between it and
+surrounding text. This is where explicit expressions come in. With explicit
+expressions you surround the variable name with parenthesis to explicitly
+show that it is an expression instead of random text.
+
+.. code-block:: c#
+
+  <h2>@(MajorVersion)@(MinorVersion)</h2>
+
+Explicit expressions are evaluated like regular c#, as such it is possible
+to render math, and much more.
+
+.. code-block:: c#
+
+  <p>@(1 + 1) = 1 + 1</p>
+
+*********
+Encoding
+*********
+Rendered data is by default encoded so that any non-text characters will be
+escaped. To render unencodedhtml use ``@Html.Raw``.
+
+.. code-block:: c#
+
+  @Html.Raw(SuperSpecialHTML)
+  @(Html.Raw(SuperSpecialHTML))
+
+``@Html.Raw`` is a Razor HTML Helper, for more information on them see
+:doc:`html-helpers`.
+
+*******
+Models
+*******
+The ``@model`` directive is a way to indicate what model the view is using. Just
+place it at the top of the view along with the class that you are wanting to
+use for the model.
+
+.. code-block:: c#
+
+  @model Example.Models.Book
+
+Later on in the document you can use implicit and explicit expressions to render
+properties and method of that model via ``@Model.SomeProperty`` or
+``@Model.SomeMethod()``.
+
+.. code-block:: c#
+
+  ISBN: @Model.ISBN
+  Author: @Model.AuthorName()
+
+The ``@using`` directive is the Razor equivalent of the c# ``using`` statement.
+
+.. code-block:: c#
+
+  @using Example.Models;
+
+************
+Code Blocks
+************
+Code blocks are a fundamental element of razor. A code block is a section where
+c# code, has been inserted directly into the html. It follows the same rules
+that a normal c# block does. Implicit expressions do not have trailing
+semicolons.
+
+.. code-block:: c#
+
+  @{
+    var foo = 123;
+  }
+
+The combination of code blocks and implicit expressions allow variables to be
+created and used.
+
+.. code-block:: c#
+
+  <div id="greeting">
+    @{
+      var GuestName = "John";
+    }
+
+    <h2>Greetings @GuestName</h2>
+  </div>
+
+You can mix plain text, and html inside of a code block by using
+html tags and the ``<text>`` tag. The ``<text>`` tag is a placeholder tag that
+is only used server side to tell razor that the following content is text and
```

Comment:
is a placeholder tag that is used to indicate that the enclosed content is text and should not be parsed as C# code.